src/main/webapp/app/shared/chart/chart-exercise-type-filter.directive.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive } from '@angular/core';
+import { ExerciseType } from 'app/entities/exercise.model';
+
+@Directive()
+export class ChartExerciseTypeFilterDirective {
+    // Ideally I would design the filter as map from ExerciseType to boolean.
+    // But I observed some unexpected casting of the ExerciseType in the ExerciseDTO
+    // that leads to the following situation: When trying to look up a value given an ExerciseType in a map with structure: ExerciseType -> boolean
+    // instead of comparing the string value of the enum, the enum key was taken as string and then used as key for the map
+    // E.g. ExerciseType.PROGRAMMING would lead to chartFilter.get('PROGRAMMING') instead of chartFilter.get('programming')
+    // This way, never a value was returned as the map did not contain such key
+    chartFilter: Map<string, boolean> = new Map();
+    numberOfActiveFilters = 0;
+    typeSet: Set<ExerciseType> = new Set();
+
+    /**
+     * Set up initial filter for the chart
+     * @param exerciseScores the score objects containing an exercise type a filter should
+     * be provided for
+     * @protected
+     */
+    protected initializeFilterOptions(exerciseScores: any[]): void {
+        this.typeSet = new Set(exerciseScores.map((score) => score.exerciseType));
+        this.typeSet.forEach((type) => {
+            this.chartFilter.set(type.toLowerCase().replace('_', '-'), true);
```

Comment:
The chain ``toLowerCase().replace('_', '-')`` is used three times in this file. I suggest to extract it into a static function instead, or just encapsulate all access to ``this.chartFilter`` into ``getChartFilter`` and ``setChartFilter`` methods (or named similarly) that include these string operations